File: XJTUTOOLBOX_CMP/androidApp/src/main/kotlin/com/xjtu/toolbox/adapter/KRRouterAdapter.kt
package com.xjtu.toolbox.adapter

import android.app.Activity
import android.content.Context
import com.tencent.kuikly.core.render.android.adapter.IKRRouterAdapter
import org.json.JSONObject

object KRRouterAdapter : IKRRouterAdapter {
    override fun openPage(
        context: Context,
        pageName: String,
        pageData: JSONObject,
    ) {
        // Single-page app: all navigation handled internally by NavigationState
        // For multi-page scenarios, start KuiklyRenderActivity here
    }

    override fun closePage(context: Context) {
        (context as? Activity)?.finish()
    }
}

File: XJTUTOOLBOX_CMP/iosApp/iosApp/Adapters/KRRouterAdapter.swift
import UIKit
import OpenKuiklyIOSRender

class KRRouterAdapter: NSObject, IKRRouterAdapter {
    func openPage(_ pageName: String, pageData: [String: Any], from context: UIViewController?) {
        let vc = KuiklyRenderViewController(pageName: pageName, pageData: pageData)
        context?.navigationController?.pushViewController(vc, animated: true)
    }

    func closePage(from context: UIViewController?) {
        context?.navigationController?.popViewController(animated: true)
    }
}
